### PR TITLE
Avoid partial name matching

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**

### CODE REVIEW

**Comment by fain182** [comment]
@AlessandroMinoccheri can you review it? (github don't permits me to ask it to you, because you created this PR)

File: src/Analyzer/PatternString.php [hunk 1]
```diff
@@ -19,8 +19,9 @@ public function matches(string $pattern): bool
             return false;
         }
 
-        if (!$this->containsWildcard($pattern) && str_starts_with($this->value, $pattern)) {
-            return true;
+        if (!$this->containsWildcard($pattern)) {
+            $slashedTerminatedPattern = str_ends_with($pattern, '\\')? $pattern : $pattern.'\\';
```
Comment (AlessandroMinoccheri) [inline src/Analyzer/PatternString.php:23]: Can we simplify these lines to explicit well the meaning?

Comment (fain182) [reply src/Analyzer/PatternString.php]: It's better now?

Comment (AlessandroMinoccheri) [reply src/Analyzer/PatternString.php]: Yes, much better @fain182

**Pull request merged**